### PR TITLE
Replace acorn/acorn-walk with esbuild (#404)

### PR DESCRIPTION
**Pull request by jahands**
Tested with this example and it generates the same routing table as before:

```javascript
// functions/api/[[path]].js
export { onRequest };
async function onRequest(context) {
  return new Response("Hello, world!");
}
export const onRequestPost = () => {};
export async function hello() {return "hello";}
```

### CODE REVIEW

**Comment by petebacondarwin** [comment]
There is a failing unit test, which seems relevant:

```
generateConfigFromFileTree
      ✕ should generate a route entry for each file in the tree (53 ms)
```

File: packages/wrangler/pages/functions/filepath-routing.ts
Comment (jahands) [reply packages/wrangler/pages/functions/filepath-routing.ts:33]: That does work better, thanks!

**Comment by jahands** [comment]
Tests should be fixed now. Wasn't handling paths correctly when calling esbuild.

File: packages/wrangler/pages/functions/filepath-routing.ts
Comment (petebacondarwin) [inline packages/wrangler/pages/functions/filepath-routing.ts:36]: What can `metafile.outputs` contain? If it is only the bundled file then there should only be one, right? If so, then I think we are good. If it can also contain all the input files that have been transpiled, then you might end up with false positives here.

File: packages/wrangler/package.json
Comment (petebacondarwin) [inline packages/wrangler/package.json:64]: Can you run `npm i` from the root of the project to ensure that `package-lock.json` is updated.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

File: packages/wrangler/pages/functions/filepath-routing.ts
Comment (jahands) [reply packages/wrangler/pages/functions/filepath-routing.ts:36]: Looks like it's only the file in the `entrypoints` that's processed here.
I'll add `bundle: false` to the build command so it's clearer that we only care about that single file's exports.

Here's an example file:
```typescript
import { onRequestDelete } from "./bar";
export function onRequestGet() {}
export function onRequestPost() {}
export async function foo() {
    return onRequestDelete()
}
```

Metafile:
```json
{
  "inputs": {
    "../../tmp/functions/foo.ts": {
      "bytes": 171,
      "imports": []
    }
  },
  "outputs": {
    "foo.js": {
      "imports": [],
      "exports": [
        "foo",
        "onRequestGet",
        "onRequestPost"
      ],
      "entryPoint": "../../tmp/functions/foo.ts",
      "inputs": {
        "../../tmp/functions/foo.ts": {
          "bytesInOutput": 172
        }
      },
      "bytes": 172
    }
  }
}
```

File: packages/wrangler/pages/functions/filepath-routing.ts
Comment (jahands) [reply packages/wrangler/pages/functions/filepath-routing.ts]: Fixed

**Comment by GregBrimble** [comment]
I'd like to review this before it gets merged. Will take a look this weekend.

File: .changeset/orange-vans-lay.md
Comment (petebacondarwin) [inline .changeset/orange-vans-lay.md:5]: Could you follow the [changeset message format](https://github.com/cloudflare/wrangler2/blob/main/CONTRIBUTING.md#changeset-message-format) here and also mention pages too, please?

File: packages/wrangler/pages/functions/filepath-routing.ts
Comment (petebacondarwin) [reply packages/wrangler/pages/functions/filepath-routing.ts:36]: Great.

**Review by petebacondarwin — APPROVED** [review]
LGTM - Please update the changeset text.

We'll wait for @GregBrimble's approval before landing.

**Review by GregBrimble — APPROVED** [review]
Looks great! Much simpler than what we had before, and another couple of dependencies removed. Great stuff!

File: .changeset/orange-vans-lay.md
Comment (jahands) [reply .changeset/orange-vans-lay.md:5]: Yes, sorry about that.

**Comment by jahands** [comment]
Got the commits cleaned up and I think I did the changeset correctly this time.

**Comment by petebacondarwin** [comment]
Great! Thanks @jahands.

**Pull request merged**
